src/reducers/AuthReducer.js

```diff
@@ -24,16 +47,46 @@ export default (state = INITIAL_STATE, action) => {
             return ({...state, appReady: true});
 
         case GUEST_TOKEN_SET:
-            return ({...state, token: payload.token, appReady: payload.appReady});
+            const guestUser = {name: payload.token};
+            return ({...state, token: payload.token, currentUser: guestUser, appReady: payload.appReady});
 
         case FACEBOOK_LOGIN_SUCCESS:
-            return ({...state, token: payload.token, appReady: payload.appReady});
+            return ({...state, token: payload.token, currentUser: payload.currentUser, appReady: payload.appReady});
 
         case FACEBOOK_LOGIN_FAILURE:
             return ({...state, token: null});
 
         case FACEBOOK_LOGOUT_COMPLETE:
-            return ({...state, token: null, appReady: false});
+            return (INITIAL_STATE);
+
+        case FAVORITE_ADDED:
+            const {favoriteToAdd} = payload;
+
+            let updatedUserWithNewFavorite = _.cloneDeep(state.currentUser);
+            updatedUserWithNewFavorite.favorites = _.concat(updatedUserWithNewFavorite.favorites, favoriteToAdd);
+
+            return ({...state, currentUser: updatedUserWithNewFavorite});
```

Comment:
When a favorite is added, we pass down the full favorite site from the action, so we can just grab that full site and add it locally to the running list of favorites.